python/lsst/cp/pipe/cpTask.py

```diff
@@ -0,0 +1,594 @@
+        # or working out what's wrong with the MONDIODE values (debug that anyway as a large difference might
+        # indicate something else going on). Note that this task doesn't really use much in its config class,
+        # but passes in things at runtime param to its run() method, hence putting in a slot for them here.
+        self.flatPairMaxPdFracDev = 0.99
+
+    def validate(self):
+        """Override of the valiate() method.
+
+        The pexConfigs of the subTasks here cannot be validated in the normal way, as they are the configs
+        for eotest, which does illegal things, and this would require an upstream PR to fix. Therefore, we
+        override the validate() method here, and use it to set the output directory for each of the tasks
+        based on the legal pexConfig parameter for the main task.
+        """
+        log = lsstLog.Log.getLogger("cp.pipe.cpTaskConfig")
+        if not self.eotestOutputPath:
+            raise RuntimeError("Must supply an output path for eotest data. "
+                               "Please set config.eotestOutputPath.")
+
+        taskList = ['fe55', 'brightPixels', 'darkPixels', 'readNoise', 'traps', 'cte', 'flatPair', 'ptc']
+        for task in taskList:
+            if getattr(self, task).output_dir != '.':
+                # Being thorough here: '.' is the eotest default. If this is not the value then the user has
+                # specified something, and we're going to clobber it, so raise a warning. Unlike to happen.
+                log.warn("OVERWRITING: Found a user defined output path of %s for %sTask. "
+                         "This has been overwritten with %s, as individually specified output paths for "
+                         "subTasks are not supported at present"%(getattr(self, task).output_dir,
+                                                                  task, self.eotestOutputPath))
+            getattr(self, task).output_dir = self.eotestOutputPath
+
+
+class CpTask(pipeBase.CmdLineTask):
+    """
+    Calibration Products Production (CP) task.
+
+    This task is used to produce the calibration products required to calibrate cameras.
+    Examples of such operations are as follows:
+    Given a set of flat-field images, find the dark pixels and columns.
+    Given a set of darks, find the bright pixels and columns.
+    Given a set of Fe55 exposures, calulate the gain of the readout chain, in e-/ADU
+    Given a set of Fe55 exposures, calulate the instrinsic PSF of the silicon, and the degradation of
+    the PSF due to CTE.
+    Given a set of flat-pairs, measure the photon transfer curve (PTC).
+    Given a set of bias frames, calculate the read noise of the system in e-.
+    Given a set of pocket-pumping exposures, find charge-traps in the silicon.
+
+    The CpTask.runEotestDirect() is only applicable to LSST sensors, and only for a specific type of dataset
```

Comment:
Because `eotest` is the analysis suite for the LSST-specific electro-optical tests run at BNL for sensor acceptance testing for the LSST project, and therefore doesn't concern itself with anything else. It is the polar-opposite of camera-agnostic.

Reply:
All the more reason to jettison `CmdLineTask`.

Reply:
Sorry, I'm not sure I follow why that adds weight to the jettisoning of `CmdLineTask`. Could you explain a little please?

Reply:
`CmdLineTask` is all about camera-agnosticism.